### PR TITLE
Add migration guide from 1.1.8 to 1.2.0

### PR DESCRIPTION
**Pull request by TheSmartnik**

### CODE REVIEW

File: UPDATING.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Upgrade / Migration Guide
+
+## Version 1.1.8 to 1.2.0
+
+### Notable Changes
+This release is all about channel options. Here a full [changelog](https://github.com/ably/ably-ruby/blob/main/CHANGELOG.md)
```
Comment (jamienewcomb) [inline UPDATING.md:6]: Minor thing
Here a full [changelog](https://github.com/ably/ably-ruby/blob/main/CHANGELOG.md) -> Here is the full [changelog](https://github.com/ably/ably-ruby/blob/main/CHANGELOG.md)

File: UPDATING.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+### Notable Changes
+This release is all about channel options. Here is the full [changelog](https://github.com/ably/ably-ruby/blob/main/CHANGELOG.md)
+
+* Channel options were extracted into a seperate model [ChannelOptions](https://github.com/ably/ably-ruby/blob/main/lib/ably/models/channel_options.rb). However it's still backward campatible with `Hash` and you don't need to do make any adjusments in your code
+
+* `ChannelOptions` now support `:params`, `:modes` and `:cipher` as options. Previosly only `:cipher` was available
+
+### Breaking Changes
+
+* Changing channel options with `Channels#get` is now deprecated
```
Comment (owenpearson) [inline UPDATING.md:14]: We should mention that it's deprecated in favour of `setOptions`

File: UPDATING.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Upgrade / Migration Guide
+
+## Version 1.1.8 to 1.2.0
+
+### Notable Changes
+This release is all about channel options. Here is the full [changelog](https://github.com/ably/ably-ruby/blob/main/CHANGELOG.md)
+
+* Channel options were extracted into a seperate model [ChannelOptions](https://github.com/ably/ably-ruby/blob/main/lib/ably/models/channel_options.rb). However it's still backward campatible with `Hash` and you don't need to do make any adjusments in your code
+
+* `ChannelOptions` now support `:params`, `:modes` and `:cipher` as options. Previosly only `:cipher` was available
```
Comment (owenpearson) [inline UPDATING.md:10]: I'd prefer "The `ChannelOptions` class now supports `:params`, `:modes` and `:cipher` as options.".

Also Previosly -> Previously

File: UPDATING.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Upgrade / Migration Guide
+
+## Version 1.1.8 to 1.2.0
+
+### Notable Changes
+This release is all about channel options. Here is the full [changelog](https://github.com/ably/ably-ruby/blob/main/CHANGELOG.md)
+
+* Channel options were extracted into a seperate model [ChannelOptions](https://github.com/ably/ably-ruby/blob/main/lib/ably/models/channel_options.rb). However it's still backward campatible with `Hash` and you don't need to do make any adjusments in your code
```
Comment (owenpearson) [inline UPDATING.md:8]: "adjusments in" -> "adjustments to"

File: UPDATING.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+* `ChannelOptions` now support `:params`, `:modes` and `:cipher` as options. Previosly only `:cipher` was available
+
+### Breaking Changes
+
+* Changing channel options with `Channels#get` is now deprecated
+
+  1. If channel is attached an exception will be raised
```
Comment (owenpearson) [inline UPDATING.md:16]: should this be 'if channel state is attached or attaching'

File: UPDATING.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  2. Otherwise warning
```
Comment (owenpearson) [inline UPDATING.md:17]: "Otherwise the library will emit a warning"

File: UPDATING.md (hunk 2 above)
Comment (owenpearson) [reply UPDATING.md:14]: and provide a small example snippet of how to change from `Channels.get` to `Channels.setOptions`

**Review by owenpearson — CHANGES_REQUESTED** [review]

File: UPDATING.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# Upgrade / Migration Guide
+
+## Version 1.1.8 to 1.2.0
+
+### Notable Changes
+This release is all about channel options. Here is the full [changelog](https://github.com/ably/ably-ruby/blob/main/CHANGELOG.md)
+
+* Channel options were extracted into a seperate model [ChannelOptions](https://github.com/ably/ably-ruby/blob/main/lib/ably/models/channel_options.rb). However it's still backward campatible with `Hash` and you don't need to do make any adjusments to your code
```
Comment (owenpearson) [inline UPDATING.md:8]: adjustments is still mispelt here

File: UPDATING.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+* The `ChannelOptions` class now supports `:params`, `:modes` and `:cipher` as options. Previously only `:cipher` was available
+
+### Breaking Changes
+
+* Changing channel options with `Channels#get` is now deprecated in favor of explicit options change
+
+  1. if channel state is attached or attaching an exception will be raised
```
Comment (owenpearson) [inline UPDATING.md:16]: Should start with a capital letter

**Review by owenpearson — CHANGES_REQUESTED** [review]

**Review by owenpearson — APPROVED** [review]
LGTM 🚀

**Review by lukaszsliwa — APPROVED** [review]

**Pull request merged**